test/common/network/listen_socket_impl_test.cc

```diff
@@ -2,43 +2,60 @@
 
 #include "common/network/utility.h"
 #include "common/network/listen_socket_impl.h"
+#include "test/test_common/network_utility.h"
 
 namespace Network {
 
-TEST(ListenSocket, All) {
-  // Test the case of a socket with given port and bind_to_port set to true.
-  TcpListenSocket socket1(uint32_t(15000), true);
+TEST(ListenSocket, BindIPv4Port) {
+  // Test the case of a socket with given address and port, and with bind_to_port set to true.
+  auto addr = Network::Test::findOrCheckFreePort("127.0.0.2:0", Address::SocketType::Stream);
+  EXPECT_LT(0U, addr->ip()->port());
+  TcpListenSocket socket1(addr, true);
   EXPECT_EQ(0, listen(socket1.fd(), 0));
-  EXPECT_EQ(15000U, socket1.localAddress()->ip()->port());
+  EXPECT_EQ(addr->ip()->port(), socket1.localAddress()->ip()->port());
+  EXPECT_EQ("127.0.0.2", socket1.localAddress()->ip()->addressAsString());
```

Comment:
I'm guessing you don't bind to zero as you want to do an explicit port in this test, so you use 127.0.0.2 with the assumption that other folks aren't listening there? If so, I would document and pick something even whackier like 127.43.27.4 which isn't likely to be something picked. We have internal environments that do you the low 127.0.0.x.